homeassistant/components/homeassistant_sky_connect/__init__.py

```diff
@@ -2,87 +2,53 @@
 
 from __future__ import annotations
 
-from homeassistant.components import usb
-from homeassistant.components.homeassistant_hardware.silabs_multiprotocol_addon import (
-    check_multi_pan_addon,
-    get_zigbee_socket,
-    multi_pan_addon_using_device,
-)
-from homeassistant.config_entries import SOURCE_HARDWARE, ConfigEntry
-from homeassistant.core import HomeAssistant, callback
-from homeassistant.exceptions import ConfigEntryNotReady, HomeAssistantError
-from homeassistant.helpers import discovery_flow
-
-from .const import DOMAIN
-from .util import get_hardware_variant, get_usb_service_info
-
-
-async def _async_usb_scan_done(hass: HomeAssistant, entry: ConfigEntry) -> None:
-    """Finish Home Assistant SkyConnect config entry setup."""
-    matcher = usb.USBCallbackMatcher(
-        domain=DOMAIN,
-        vid=entry.data["vid"].upper(),
-        pid=entry.data["pid"].upper(),
-        serial_number=entry.data["serial_number"].lower(),
-        manufacturer=entry.data["manufacturer"].lower(),
-        description=entry.data["description"].lower(),
-    )
+import logging
 
-    if not usb.async_is_plugged_in(hass, matcher):
-        # The USB dongle is not plugged in, remove the config entry
-        hass.async_create_task(
-            hass.config_entries.async_remove(entry.entry_id), eager_start=True
-        )
-        return
-
-    usb_dev = entry.data["device"]
-    # The call to get_serial_by_id can be removed in HA Core 2024.1
-    dev_path = await hass.async_add_executor_job(usb.get_serial_by_id, usb_dev)
-
-    if not await multi_pan_addon_using_device(hass, dev_path):
-        usb_info = get_usb_service_info(entry)
-        await hass.config_entries.flow.async_init(
-            "zha",
-            context={"source": "usb"},
-            data=usb_info,
-        )
-        return
-
-    hw_variant = get_hardware_variant(entry)
-    hw_discovery_data = {
-        "name": f"{hw_variant.short_name} Multiprotocol",
-        "port": {
-            "path": get_zigbee_socket(),
-        },
-        "radio_type": "ezsp",
-    }
-    discovery_flow.async_create_flow(
-        hass,
-        "zha",
-        context={"source": SOURCE_HARDWARE},
-        data=hw_discovery_data,
-    )
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .util import guess_firmware_type
+
+_LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
The old implementation did two things:

1. `check_multi_pan_addon` which essentially started the Multi-Protocol add-on whenever it was installed. This has some problems, like if multiple SkyConnects are used, or if the user migrated away while still having the Multi-Protocol add-on installed. As we recommend dedicated Thread firmware nowadays, Multi-Protocol is seen as self managed (the user should control the add-on, e.g. make sure it is configured to auto start etc.).
2. The USB scan created a ZHA discovery for the Multi-Protocol add-on. For now, the user will have to manually configure the ZHA integration when using the Multi-Protocol add-on. The plan is to add a hassio discovery to make this process automatic again, so it will be quite easy again to install Multi-Protocol.

So essentially, for the current focus on Thread/Zigbee, these Multi-Protocol related things are no longer required.